bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/StreamingJsonValue.java

```diff
@@ -150,6 +151,20 @@ void appendToCache(Object value) {
         super.add(size, value);
     }
 
+    @Override
+    public String getJSONString() {
+        // Consume and materialize the stream.
+        buildDatasource();
```

Comment:
Wouldn't it be better to check if `datasource.hasNext()` and then do `buildDatasource()` like it has been done for the others?

Reply:
buildDatasource does that right.

Reply:
Just that we'll be avoiding the try-catch block but in this scenario that shouldn't be much of a concern I guess. 

+1 to keep as is, since with the if check we seem to be duplicating the hasNext() check anyway.